tests/storage/fsp/conftest.py

```diff
@@ -10,12 +10,13 @@
 DIRECTORIES_PATH = 'directories'
 
 @pytest.fixture(scope='package')
-def host_with_runx_repo(host_with_saved_yum_state):
-    host = host_with_saved_yum_state
-    host.add_xcpng_repo(FSP_REPO_NAME, 'vates')
-    yield host
+def host_with_runx_repo(host):
+    with __host_with_saved_yum_state_cm as host:
+        host.add_xcpng_repo(FSP_REPO_NAME, 'vates')
+        yield host
     # teardown
-    host.remove_xcpng_repo(FSP_REPO_NAME)
+        host.remove_xcpng_repo(FSP_REPO_NAME)
+    host.restart_toolstack(verify=True)
```

Comment:
I'm not sure why `host_with_runx_repo` depends on `host_with_saved_yum_state' , as itself does not make a change to the yum state.  That causes this toolstack restart to be strangely placed.